cogment_verse/web/web_app/src/controls/AtariPongPzFeedback.jsx

```diff
@@ -0,0 +1,185 @@
+import { serializePlayerAction, DType, Space } from "../utils/spaceSerialization";
+
+const ACTION_SPACE = new Space({
+    box: {
+        low: {
+            // Only the dtype and shape of the lower bound are used to serialize the action
+            dtype: DType.DTYPE_FLOAT32,
+            shape: [1],
+        },
+    },
+});
+
+export const AtariPongPzHfbEnvironments = ["environments.pettingzoo_atari_adapter.HumanFeedbackEnvironment/pettingzoo.atari.pong_v3"];
+export const AtariPongPzFeedback = ({ sendAction, fps = 30, actorClass, observation, ...props }) => {
+    const [paused, setPaused] = useState(false);
+    const [humanMode, setHumanMode] = useState(true);
+    const [selectedFeedback, setFeedback] = useState(2);
+    const [humanTurn, setHumanTurn] = useState(false);
+    const [playerPos, setPlayerPos] = useState('');
+    const [playerNameDisplay, setPlayerNameDisplay] = useState(false);
+    const playerName = observation?.gamePlayerName;
+    const timeStep = observation?.step;
+    const feedbackRequired = observation?.feedbackRequired;
+    const gymAction = observation?.actionValue
+    const actionList = ['NONE', 'FIRE', "UP", 'DOWN', "FIRE UP", " FIRE DOWN"]
+
+
+    useEffect(() => {
+        if (playerName) {
+            if (playerName.includes('first')) {
+                setPlayerPos('right');
+            } else {
+                setPlayerPos('left');
+            }
+        }
+    }, [playerName]);
+
+    useEffect(() => {
+        if (feedbackRequired) {
+            setHumanTurn(true);
+        } else {
+            setHumanTurn(false);
+        }
+    }, [feedbackRequired]);
+
+    useEffect(() => {
+        if (!humanMode || !humanTurn || timeStep % 3 != 0) {
+            setPlayerNameDisplay(false);
+        } else {
+            setPlayerNameDisplay(true);
+        }
+    }, [timeStep, humanMode, humanTurn]);
+
+    const togglePause = useCallback(() => setPaused((paused) => !paused), [setPaused]);
+    const toggleHuman = useCallback(() => setHumanMode((humanMode) => !humanMode), [setHumanMode]);
+    useDocumentKeypressListener("p", togglePause);
+    useDocumentKeypressListener("h", toggleHuman);
+
+    const { currentFps } = useRealTimeUpdateV2(fps, paused);
+
+    if (!humanMode || !humanTurn || timeStep % 3 != 0) {
+        // Neural action 
+        const action = serializePlayerAction(ACTION_SPACE, [0.0]);
+        sendAction(action);
+    }
+    const sendFeedback = (feedback) => {
+
+        let action;
+        if (feedback == "LIKE") {
+            setFeedback(1);
+            action = serializePlayerAction(ACTION_SPACE, [1.0]);
+        } else if (feedback == "DISLIKE") {
+            setFeedback(-1);
+            action = serializePlayerAction(ACTION_SPACE, [-1.0]);
+        } else {
+            setFeedback(0);
+            action = serializePlayerAction(ACTION_SPACE, [0.0]);
+        }
+        setTimeout(() => { setFeedback(2) }, 150);
+        sendAction(action);
+    };
```

Comment:
just use add_reward, rewards shouldn't be encoded in actions

Reply:
I would be great If you have an example or doc showing how to do it